### PR TITLE
WIP: [CFM-1554-fix] Allocate correct permissions to Node Gallery API functionality.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-1554

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 1]
```diff
@@ -224,6 +224,28 @@ function c4m_og_menu() {
 function c4m_og_menu_alter(&$items) {
   // Adding a member should be allow to SA only (CFM-1292).
   $items['group/%/%/admin/people/add-user']['access callback'] = 'c4m_og_add_member_to_group_access';
+
+  // Assign access to Node Gallery API features, if user has permission to
+  // edit the gallery.
+  $node_galery_api_features = array('upload', 'items', 'manage', 'sort');
+  foreach ($node_galery_api_features as $suffix) {
+    $path = 'node/%node_gallery_api_gallery/' . $suffix;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: let's drop the `$path` and use it directly in the `$items` variable (more readable for `hook_menu_alter`).

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 2]
```diff
@@ -224,6 +224,28 @@ function c4m_og_menu() {
...
+  }
+
+  // Assign access to upload Node Gallery API item types, per upload access
+  // function.
+  // Node Gallery API item types are resolve in same manner
+  // as at node_gallery_api_menu().
+  $relationship_types = node_gallery_api_get_relationship_type(NULL, NULL, NULL, TRUE);
+  foreach ($relationship_types as $relationship) {
+    foreach ($relationship->item_types as $item_type) {
+      $path = 'node/%node_gallery_api_gallery/upload/' . $item_type;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: let's drop the `$path` and use it directly in the `$items` variable (more readable for `hook_menu_alter`).

**Comment by anvmn** [comment]
Closing in favor of #1521

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module (hunk 1 above)
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: Fixed here: https://github.com/capacity4dev/capacity4more/pull/1521/commits/d360fe245773bfb0af8189f8edc043e7172e6e16

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module (hunk 2 above)
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: Fixed here: https://github.com/capacity4dev/capacity4more/pull/1521/commits/d360fe245773bfb0af8189f8edc043e7172e6e16

**Pull request closed without merging**